src/dune/build.ml

```diff
@@ -120,11 +127,13 @@ let if_file_exists p ~then_ ~else_ =
 let file_exists p = if_file_exists p ~then_:(return true) ~else_:(return false)
 
 let file_exists_opt p t =
-  if_file_exists p ~then_:(t >>^ Option.some) ~else_:(return None)
+  if_file_exists p ~then_:(map ~f:Option.some t) ~else_:(return None)
 
 let paths_existing paths =
-  List.fold_left paths ~init:(return true) ~f:(fun acc file ->
-      if_file_exists file ~then_:(path file) ~else_:(return ()) >>> acc)
+  ignore
```

Comment:
This `ignore` is a bit confusing because it's easy to think of it as `'a -> unit`. 
In jane we started renaming such ignores to `ignore_m`.

Reply:
We could be more explicit here as follows:
```ocaml
let paths_existing paths =
  let+ (_ : unit list) =
    all
      (List.map paths ~f:(fun file ->
           if_file_exists file ~then_:(path file) ~else_:(return ())))
  in
  ()
```

Reply:
Done